camel-core/src/main/java/org/apache/camel/language/simple/ast/SimpleFunctionExpression.java

```diff
@@ -155,13 +155,22 @@ private Expression createSimpleExpression(String function, boolean strict) {
         // date: prefix
         remainder = ifStartsWithReturnRemainder("date:", function);
         if (remainder != null) {
-            String[] parts = remainder.split(":");
-            if (parts.length < 2) {
-                throw new SimpleParserException("Valid syntax: ${date:command:pattern} was: " + function, token.getIndex());
```

Comment:
The test doesn't make sense anymore because the pattern part is now optional. Possible extraneous parts are ignored as before.